corehq/apps/users/models.py

```diff
@@ -1659,62 +1659,88 @@ def set_location(self, location):
             location.group_id
         })
 
-        self.location_id = location.location_id
         self.update_fixture_status(UserFixtureType.LOCATION)
+        self.location_id = location.location_id
         self.get_domain_membership(self.domain).location_id = location.location_id
         if self.location_id not in self.assigned_location_ids:
-            self.reset_locations(self.assigned_location_ids + [self.location_id])
-        else:
-            self.save()
+            self.assigned_location_ids.append(self.location_id)
+            self.get_domain_membership(self.domain).assigned_location_ids.append(self.location_id)
+            self.user_data['commcare_location_ids'] = ','.join(self.assigned_location_ids)
+        self.save()
 
     def unset_location(self):
         """
-        Unset the primary location and remove all associated user data and cases
+        Resets primary location to next available location from assigned_location_ids.
+            If there are no more locations in assigned_location_ids,
+            then the primary location and user data are cleared
```

Comment:
This seems like confusing behavior, what's the idea behind it?  Would it be possible to remove this function entirely and instead use `unset_location_by_id` and something like `unset_all_locations` or `reset_locations([])`?